src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,444 @@
+                            draft.getSku(), SkuKeyTuple.of(draft).getKey()), ex);
+                        return null;
+                    });
+            }
+        } else {
+            statistics.incrementFailed();
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    /**
+     * Tries to create Inventory Entry in CTP project, using {@code draft}.
+     * Before calling API, the channel reference from {@code draft} is replaced, so it points to proper channel ID
+     * in target system.
+     * It either creates inventory entry and increments "created" counter in statistics, or increments "failed" counter
+     * and executes error callback function in case of any exception.
+     *
+     * @param draft draft of new inventory entry
+     * @return {@link CompletionStage} instance that indicates method progress
+     */
+    @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION") // https://github.com/findbugsproject/findbugs/issues/79
+    private CompletionStage<Void> attemptCreate(final InventoryEntryDraft draft) {
+        final Optional<InventoryEntryDraft> fixedDraft = replaceChannelReference(draft);
+        if (fixedDraft.isPresent()) {
+            return inventoryService.createInventoryEntry(fixedDraft.get())
+                .thenAccept(createdEntry -> statistics.incrementCreated())
+                .exceptionally(ex -> {
+                    statistics.incrementFailed();
+                    handleFailure(format("Failed to create inventory entry of sku '%s' and supply channel key '%s'",
+                        draft.getSku(), SkuKeyTuple.of(draft).getKey()), ex);
+                    return null;
+                });
+        } else {
+            statistics.incrementFailed();
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    /**
+     * Returns {@link Optional} that may contain {@link InventoryEntryDraft}. The payload of optional would be:
+     * <ul>
+     *     <li>Same {@code draft} instance if it has no reference to supply channel</li>
+     *     <li>New {@link InventoryEntryDraft} instance if {@code draft} contains reference to supply channel.
+     *     New instance would have same values as {@code draft} except for supply channel reference. Reference
+     *     will be replaced with reference that points to ID of existing channel for key given in draft.</li>
+     *     <li>Empty if supply channel for key wasn't found in {@code supplyChannelKeyToId}</li>
+     * </ul>
+     *
+     * @param draft inventory entry draft from processed list
+     * @return {@link Optional} with draft that is prepared to being created or compared with existing
+     * {@link InventoryEntry}, or empty optional if method fails to find supply channel ID that should be referenced
+     */
+    private Optional<InventoryEntryDraft> replaceChannelReference(final InventoryEntryDraft draft) {
+        final String supplyChannelKey = SkuKeyTuple.of(draft).getKey();
```

Comment:
I mean that this [line](https://github.com/commercetools/commercetools-sync-java/pull/19/files#diff-acd7a55d6d411aec41968e0c118dc24fR48) could be mistake for `InventoryEntry` with not expanded reference. For 2 `InventoryEntries` with same SKU where the first one have not expanded reference to supply channel and the second have no reference at all, the same `SkuKeyTuple` would be returned (of sku = sku and key = null), but such `SkuKeyTuple` should represend only the second entry. But as I said later, such situation will not occur in current implementation.